distro_test.go

```diff
@@ -140,16 +146,22 @@ configuration:
     - LANG: en_US.UTF-8
     - PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
     - TERM: xterm-256color
-`, realDistro.Name())},
+`, realDistro.Name(), realID),
+		},
 		"fake distro": {
 			distro: &fakeDistro,
-			wants: fmt.Sprintf(`distro: %s
-configuration: error in GetConfiguration: failed syscall to WslGetDistributionConfiguration
-`, fakeDistro.Name())},
+			wants: fmt.Sprintf(`name: %s
+guid: distro is not registered
+configuration: |
+  error in GetConfiguration: failed syscall to WslGetDistributionConfiguration
+`, fakeDistro.Name()),
+		},
 		"wrong distro": {
```

Comment:
I’mo pretty sure if you look at the code coverage (which ofc, you do, right? :)), this is not covered:
* d.IsRegistered() failing
* d.GUID() failing

Is there any way to have them failing without being in the "unregistered" distro case?

Reply:
They can fail only for Windows' related issues:
- If a syscall to the Win32 API fails
- If the registry structure is not the one expected.

Depending on what a final application looks like, running it could fail cryptically on older versions of Windows where WSL is not available. Then the Lxss key in the registry would not exist.
If somehow the key exists, but the wslAPI.dll is not found, then it panics.

In the first case, the error looks like:
```
failed to detect if "Ubuntu" is registered: failed to obtain list of registered distros: failed to open lxss registry: The system cannot find the file specified.
```
If somehow access to the WSL API is teh first thing to fail, it would panic like:
```
%!v(PANIC=String method: Failed to load wslapi.dll: The specified module could not be found.)
Disable windows paths and fail to run notepad:
panic: Failed to load wslapiA.dll: The specified module could not be found.

(Stack trace here)
```

Reply:
The panics are due to the implementation of the `syscall` module, but they could be wrapper with a recover if we wanted to.

Reply:
These errors could be exercised if I dedicated some work to this (now ancient) PR: https://github.com/ubuntu/GoWSL/pull/9

Reply:
> Is there any way to have them failing without being in the "unregistered" distro case?

There's also the "name cannot be converted to UTF16" case, which is covered (it only fails when there's a null character in the name.

Reply:
This is where the tests in GoWSL are a little bit too much of integration tests. Normally, we would inject a fixture with a mock registry to make the API call failing for instance and exercise the error case.

But I’m unsure to understand the "wrong distro" case, it says that the distro is not registered, so, it doesn’t cover "return id, fmt.Errorf("error accessing the registry to obtain distro GUID: %v", err)" for instance?

Reply:
> But I’m unsure to understand the "wrong distro" case, it says that the distro is not registered

You're right, I forgot that IsRegistered doesn't use the Win32 api and goes straight to the registry.

At the moment there is no way of mocking the registry in the tests, but I agree that we should find a way to inject a mock registry and Win32 API. I belive this would be better left to a future PR since extracting these dependencies would require some major refactor in places unrelated to the GUID.

Reply:
agreed :)